src/lib/optimizer/strategy/index_scan_rule.cpp

```diff
@@ -86,6 +82,11 @@ bool IndexScanRule::_is_index_scan_applicable(const ChunkIndexStatistics& index_
     return false;
   }
 
+  if (!(operator_predicate.predicate_condition == PredicateCondition::Equals ||
```

Comment:
just a question of flavor, but apply De Morgan's law?

Reply:
It has been a long time since the I used De Morgan's law.
Good catch!